2-promises/solution.js

```diff
@@ -25,15 +25,34 @@ const id = yourRandomMethod() //third run
 */
 
 function solution() {
-    // YOUR SOLUTION GOES HERE
-
-    // You generate your id value here
-
-    // You call the lastnames method with your id
-
-    // You call the firstname method
-
-    // You log the fullname, or error, here
+  // YOUR SOLUTION GOES HERE
+
+  // You generate your id value here
+  const randomId = () => {
+    const random = Math.floor(Math.random() * 3);
+    if (random == 1) {
+      return Math.floor(Math.random() * 100);
+    } else if (random == 2) {
+      return -random;
+    }
+    return true;
+  };
+
+  // You call the lastnames method with your id
+  const getLastnameWithId = require("./lastnames");
+  getLastnameWithId(randomId())
+    .then((lastname) => {
+      // You call the firstname method
+      const getFirstnameWithLastname = require("./firstnames");
+      return Promise.all([getFirstnameWithLastname(lastname), lastname]);
```

Comment:
i think we could omit the Promise.all here, why do we need it?

Reply:
to pass lastName to the next then, we could just declare lastName and firstName at the begining of the function